[REFACTOR] Enhancements and Code Cleanup

DeekshithSH opened the pull request:
- Use fork of Pyrogram (fork by KurimuzonAkuma).
- Show Stream Button when the message contains an audio or video file.
- Changed import order.
- Removed unused imports and trailing spaces.
- Used lazy formatting in logging.
- Added type hinting.
- Use Rotating Logs
- Don't send body for Head Method

DeekshithSH committed:
Use fork of pyrogram

DeekshithSH committed:
Show Stream Button if Message is Audio or Video

DeekshithSH committed:
Removed shortened link

DeekshithSH committed:
Changed import order
Removed Unused imports, Trailing spaces
Use Lazy formating in logging
Type Hinting

EverythingSuckz reviewed (changes requested):
Thank you for your contributions. I've found some small issues.

DeekshithSH committed:
Revert "Removed shortened link"

This reverts commit bcf0d73af7830dc68ed8c0d3714f22ad702fbb1a.

DeekshithSH committed:
use dict

DeekshithSH commented on requirements.txt:
```diff
@@ -1,4 +1,4 @@
-pyrogram<=2.0.93 
-python-dotenv<=0.20.0
+aiohttp<=3.11.14
+kurigram<=2.1.39
```
Hydrogram is also good, but its last commit was 4 months ago, so I wasn't sure if it's still actively maintained. Kurigram, on the other hand, seems to be receiving regular updates.

EverythingSuckz replied:
Alright, I'll look into it 👍

DeekshithSH committed:
Implement Rotating Logs

Use rotating logs to prevent the log file from growing indefinitely and consuming excessive storage.

EverythingSuckz reviewed (approved):
LGTM!
I'll merge this as soon as I test this code

DeekshithSH committed:
Don't load body if HTTP method is HEAD

pranjaleholo-beep commented:
hey , brother , can you implement UPNP support , i have  a perfect .py for your old python project , i do not know go . plz make 
[upnp.py](https://github.com/user-attachments/files/25522219/upnp.py)
[upnp.py](https://github.com/user-attachments/files/25522219/upnp.py)
[upnp.py](https://github.com/user-attachments/files/25522219/upnp.py)
